### DISCUSSION TITLE
Card used to increase memory?

### DISCUSSION THREAD

**Discussion by NapierDeltic** [suchmememanyskill/CYD-Klipper#91] [Ideas]
Hi suchmememanyskill. First of all thanks for this excellent development!
Question from a complete newbie: SD card cannot be used to increase stored data capacity (thumbnail case discussed on reddit)?

**Comment by suchmememanyskill** [comment]
Technically, yes, it would help in the memory department
I don't think that's very feasible though. 

First of all, if i want to go down the route of hyper optimalisations, i could write my own png decoder, and have it load line by line (and somehow make this work with lvgl), but this is insanely difficult and i don't think it's worth the effort.

If i had an sd, yes, i could slowly decode the .png, then let lvgl load in the bitmap (which still takes a huge amount of memory(!))

Second, png is really not an ideal for this. JPG or QOI would be way better for this. Sadly Moonraker doesn't support JPG or QOI, so i can't use them

**Reply by NapierDeltic** [reply]
Thanks for clarification! Can't wait for CYD delivery :)!
